### PR TITLE
First draft with RX window lead-time and buffer

### PR DESCRIPTION
**Pull request by lthiery**

### CODE REVIEW

**Comment by lthiery** [comment]
@lucasgranberg

File: lora-phy/Cargo.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ rustdoc-args = ["--cfg", "docsrs"]
 
 [dependencies]
 defmt = { version = "0.3" }
-lora-modulation = { version = ">=0.1.2" }
+lora-modulation = { path = "../lora-modulation", version = ">=0.1.2" }
```
Comment (lucasgranberg) [inline lora-phy/Cargo.toml:17]: does version make sense with path?

File: lora-phy/src/lorawan_radio.rs [hunk 2]
```diff
@@ -34,8 +33,8 @@ where
         Self {
             lora,
             rx_pkt_params: None,
-            rx_window_offset_ms: DEFAULT_RX_WINDOW_OFFSET_MS,
-            rx_window_duration_ms: DEFAULT_RX_WINDOW_DURATION_MS,
+            rx_window_lead_time: DEFAULT_RX_WINDOW_LEAD_TIME,
+            rx_window_buffer: DEFAULT_RX_WINDOW_LEAD_TIME,
```
Comment (lucasgranberg) [inline lora-phy/src/lorawan_radio.rs:37]: same const for both?

**Review by lucasgranberg — COMMENTED** [review]
I think it makes sense to interact with the mac layer directly instead of handling the region. In my mind Region should be constant. Channel plans needs to be configurable and are kind of part of region so some care should be taken with them. I think the changes ceekdee did to my lorawan code could work as inspiration.
https://github.com/lucasgranberg/lorawan/blob/main/src/mac/region/mod.rs

**Pull request merged**